### ISSUE TITLE
Crashing game

### ISSUE THREAD

**Issue by number117** [svenhjol/Charm#773]
Version 4.0.0 and now 4.1.0 is crashing my game on startup.

Any known issues with my other mods??

"C:\Users\hej\AppData\Roaming\.minecraft\mods\Adorn-3.4.0+1.18.2-fabric.jar"
"C:\Users\hej\AppData\Roaming\.minecraft\mods\architectury-4.1.36.jar"
"C:\Users\hej\AppData\Roaming\.minecraft\mods\blockus-2.4.3+1.18.2.jar"
"C:\Users\hej\AppData\Roaming\.minecraft\mods\Bookshelf-Fabric-1.18.2-13.0.7.jar"
"C:\Users\hej\AppData\Roaming\.minecraft\mods\charm-fabric-1.18-4.1.0.jar"
"C:\Users\hej\AppData\Roaming\.minecraft\mods\cloth-api-3.2.64.jar"
"C:\Users\hej\AppData\Roaming\.minecraft\mods\cloth-config-6.2.57-fabric.jar"
"C:\Users\hej\AppData\Roaming\.minecraft\mods\custom-crosshair-mod-v1.3.9-fabric-mc1.18.2.jar"
"C:\Users\hej\AppData\Roaming\.minecraft\mods\EnchantmentDescriptions-Fabric-1.18.2-10.0.2.jar"
"C:\Users\hej\AppData\Roaming\.minecraft\mods\fabric-api-0.48.0+1.18.2.jar"
"C:\Users\hej\AppData\Roaming\.minecraft\mods\fabric-installer-0.10.2.jar"
"C:\Users\hej\AppData\Roaming\.minecraft\mods\fabric-language-kotlin-1.7.1+kotlin.1.6.10.jar"
"C:\Users\hej\AppData\Roaming\.minecraft\mods\feature_nbt_deadlock_be_gone_fabric-2.0.0+1.18.2-rc1.jar"
"C:\Users\hej\AppData\Roaming\.minecraft\mods\Gobber2-Fabric-1.18.2-2.5.68.jar"
"C:\Users\hej\AppData\Roaming\.minecraft\mods\infusion-table-mod-1.0.5.jar"
"C:\Users\hej\AppData\Roaming\.minecraft\mods\InventorySorter-1.8.6-1.18.jar"
"C:\Users\hej\AppData\Roaming\.minecraft\mods\megane-7.3.0.jar"
"C:\Users\hej\AppData\Roaming\.minecraft\mods\modmenu-3.1.1.jar"
"C:\Users\hej\AppData\Roaming\.minecraft\mods\morevillagers-FABRIC-1.18.2-3.2.0.jar"
"C:\Users\hej\AppData\Roaming\.minecraft\mods\optifabric-1.13.0.jar"
"C:\Users\hej\AppData\Roaming\.minecraft\mods\OptiFine_1.18.2_HD_U_H6.jar"
"C:\Users\hej\AppData\Roaming\.minecraft\mods\RebornCore-5.2.0-beta.2.jar"
"C:\Users\hej\AppData\Roaming\.minecraft\mods\repurposed_structures_fabric-4.1.1+1.18.2.jar"
"C:\Users\hej\AppData\Roaming\.minecraft\mods\RoughlyEnoughItems-8.0.442.jar"
"C:\Users\hej\AppData\Roaming\.minecraft\mods\SimpleBackpack_Fabric-1.18.2-1.2.33.jar"
"C:\Users\hej\AppData\Roaming\.minecraft\mods\TechReborn-5.2.0-beta.2.jar"
"C:\Users\hej\AppData\Roaming\.minecraft\mods\trinkets-3.3.0.jar"
"C:\Users\hej\AppData\Roaming\.minecraft\mods\wraith-silkspawners-1.1.1.jar"
"C:\Users\hej\AppData\Roaming\.minecraft\mods\wthit-fabric-4.8.0.jar"
"C:\Users\hej\AppData\Roaming\.minecraft\mods\Xaeros_Minimap_22.6.1_Fabric_1.18.2.jar"
"C:\Users\hej\AppData\Roaming\.minecraft\mods\XaerosWorldMap_1.20.6_Fabric_1.18.2.jar"

**Comment by chland** [comment]
Just out of curiosity - did you disable the AmethystNoteBlock ?If yes, try re-enabling it in the config-file. I had crashes on startup, too and this fixed it for me.

**Comment by number117** [comment]
No, didn't disable anything.
Just downloaded the mod, put it into my modfolder, and started the game and it crashed. repeatedly
So i deleted the mod and started the game, no problem.

**Comment by UntalentedAmateur** [comment]
We've had serious issues with both 4.0.0 and 4.1.0 crashing our local and remote games without anything useful appearing in the logs (local) or console (remote).

What we were able to find, after a LOT of fault finding, is the following:

- Client and Server both need to have Charm installed.
- The same version must be on both
- The same configuration settings (/config/charm.toml) need to be present on both client and server or else it will crash / prevent login
- If you're running Charm on multiple servers, with multiple configs, you WILL need multiple client configs that match them  in order to successfully connect to them.  The solution we found worked best here was to have dedicated cloned installations, with their own mod folders (game folder selection in the vanilla launcher), which also meant they had their own config folders as well.
